runtime/app-runtime.js

```diff
@@ -587,32 +601,15 @@ AppRuntime.prototype.resetServices = function resetServices (options) {
   return Promise.all(promises)
 }
 
-AppRuntime.prototype.appPause = function appPause (appId) {
-  logger.info('Pausing resources of app', appId)
-  var promises = [
-    this.component.light.stopSoundByAppId(appId),
-    this.multimediaMethod('pause', [ appId ])
-  ]
-  return Promise.all(promises)
-    .catch(err => logger.error('Unexpected error on pausing resources of app', appId, err.stack))
-}
-
-AppRuntime.prototype.appGC = function appGC (appId) {
+AppRuntime.prototype.appDidExit = function appDidExit (appId) {
   logger.info('Collecting resources of app', appId)
   var promises = [
-    this.component.light.stopByAppId(appId),
-    this.component.light.stopSoundByAppId(appId),
     this.multimediaMethod('stop', [ appId ]),
-    this.ttsMethod('stop', [ appId ])
+    this.ttsMethod('stop', [ appId ]),
+    this.componentsInvoke('appDidExit', [ appId ]),
+    this.descriptorsInvoke('appDidExit', [ appId ])
```

Comment:
Invokes the descriptors looks a little bit weird, could we move the keyboard descriptor to its component?